cloudinit/sources/helpers/vmware/imc/config_nic.py

```diff
@@ -274,7 +274,7 @@ def configure(self, osfamily=None):
         lines = [
             "# DO NOT EDIT THIS FILE BY HAND --"
             " AUTOMATICALLY GENERATED BY cloud-init",
-            "source /etc/network/interfaces.d/*.cfg",
+            "source /etc/network/interfaces.d/*",
```

Comment:
I realize now that both Ubuntu and Debian are of the same osfamily.  We will need to include both patterns instead of one or the other.

```
source /etc/network/interfaces.d/*.cfg
source-directory /etc/network/interfaces.d
```

On Ubuntu distro, cloud-init writes /etc/network/interfaces.d/50-cloud-init.cfg
On Debian distro, cloud-init writes /etc/network/interfaces.d/50-cloud-init

sourcing both elements should pick up either file correctly.